src/rmutil/periodic.h

```diff
@@ -39,6 +40,19 @@ void RMUtilTimer_SetInterval(struct RMUtilTimer *t, struct timespec newInterval)
  */
 int RMUtilTimer_Terminate(struct RMUtilTimer *t);
 
+/*
+ * This function cause the callback function to execute with out waiting for timetout.
+ * It does it by sending a signal to the conditional variable causing the pthread_cond_timedwait
+ * to return. It receive a blocked client which added to the blocks client chain.
+ * After finish executing the cb the block client is unblocked.
+ *
+ * We use this to allow running gc with ft.debug FORCE_GCINVOKE and receive a 'done' reply when
+ * the gc finished its run.
+ */
+void RMUtilTimer_ForceInvoke(struct RMUtilTimer *t, RedisModuleBlockedClient *bClient);
```

Comment:
Why does the list of clients needs to be inside the timer's setting and not the GC's setting (i.e. the GC object)? It would be better if this were all done inside *there* (make a list in the gc context; whenever the gc timer callback is invoked, see if there are any outstanding blocked clients). If you want to invoke immediately, you can certainly add a function for that.

Reply:
Invoke immediate is problematic because gc tries to acquire the lock, if you call it from the main thread you get a deadlock, it is possible to add the chain to the gc object, I agree and will be done.

Reply:
Rethinking on it I do not see why not adding a force invoke capability to RMUtilTimer which means that it will invoke the callback with out waiting from the timeout to end. I think it is ok.

Reply:
`Signal` is already this force-invoke capability. The issue here is with the blocked client, that kind of logic should go elsewhere (handle the blocked clients elsewhere)

Reply:
handling it now ...